kotlin/find-address/src/main/java/com/esri/arcgisruntime/sample/findaddress/MainActivity.kt

```diff
@@ -0,0 +1,342 @@
+   */
+  private fun setupAddressSearchView() {
+    addressGeocodeParameters = GeocodeParameters().apply {
+      // get place name and street address attributes
+      resultAttributeNames.addAll(listOf("PlaceName", "StAddr"))
+      // return only the closest result
+      maxResults = 1
+
+      addressSearchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+        override fun onQueryTextSubmit(address: String): Boolean {
+          // geocode typed address
+          geoCodeTypedAddress(address)
+          // clear focus from search views
+          addressSearchView.clearFocus()
+          return true
+        }
+
+        override fun onQueryTextChange(newText: String): Boolean {
+         // if the newText string isn't empty, get suggestions from the locator task
+          if (newText.isNotEmpty()){
+            val suggestionsFuture = locatorTask.suggestAsync(newText)
+            suggestionsFuture.addDoneListener{
+              try {
+                // get the results of the async operation
+                val suggestResults = suggestionsFuture.get()
+                // set up parameters for searching with MatrixCursor
+                val address = "address"
+                val columnNames = arrayOf(BaseColumns._ID, address)
+                val suggestionsCursor = MatrixCursor(columnNames)
+
+                // add each address suggestion to a new row
+                for ((key, result) in suggestResults.withIndex()) {
+                  suggestionsCursor.addRow(arrayOf<Any>(key, result.label))
+                }
+                // column names for the adapter to look at when mapping data
+                val cols = arrayOf(address)
+                // ids that show where data should be assigned in the layout
+                val to = intArrayOf(R.id.suggestion_address)
+                // define SimpleCursorAdapter
+                val suggestionAdapter = SimpleCursorAdapter(this@MainActivity,
+                  R.layout.suggestion, suggestionsCursor, cols, to, 0)
+
+                addressSearchView.suggestionsAdapter = suggestionAdapter
+                // handle an address suggestion being chosen
+                addressSearchView.setOnSuggestionListener(object: SearchView.OnSuggestionListener{
+                  override fun onSuggestionSelect(position: Int): Boolean { return false }
+                  override fun onSuggestionClick(position: Int): Boolean {
+                    // get the selected row
+                    val selectedRow = suggestionAdapter.getItem(position) as MatrixCursor
+                    // get the row's index
+                    val selectedCursorIndex = selectedRow.getColumnIndex(address)
+                    // get the string from the row at index
+                    val selectedAddress = selectedRow.getString(selectedCursorIndex)
+                    // use clicked suggestion as query
+                    addressSearchView.setQuery(selectedAddress, true)
```

Comment:
```suggestion
                    // get the selected row
                    (suggestionAdapter.getItem(position) as? MatrixCursor)?.let { selectedRow ->
                      // get the row's index
                      val selectedCursorIndex = selectedRow.getColumnIndex(address)
                      // get the string from the row at index
                      val selectedAddress = selectedRow.getString(selectedCursorIndex)
                      // use clicked suggestion as query
                      addressSearchView.setQuery(selectedAddress, true)
                    }
```

A bit belts and braces, but can do a check on the cast within a let